client/pages/topsecret.jsx

```diff
@@ -13,31 +63,125 @@ const HomePage = React.createClass({
       console.log(this.context);
     }
 
+
   },
 
   render() {
```

Comment:
The render function is huge! Try calling other functions that return JSX instead of doing it all in here with if/else statements.

For example:

```
render() {
   if (this.bankAccountIsEmpty()) {
      return this.renderCannotPurchaseMoreTools();
   }

   return this.renderToolPurchaseInput()
}
```